substrate/primitives/core/src/ecdsa.rs

```diff
@@ -96,18 +97,27 @@ impl Public {
 	/// Create a new instance from the given full public key.
 	///
 	/// This will convert the full public key into the compressed format.
-	#[cfg(feature = "std")]
 	pub fn from_full(full: &[u8]) -> Result<Self, ()> {
-		let pubkey = if full.len() == 64 {
+		let mut tagged_full = [0u8; 65];
+		let full = if full.len() == 64 {
 			// Tag it as uncompressed public key.
-			let mut tagged_full = [0u8; 65];
 			tagged_full[0] = 0x04;
 			tagged_full[1..].copy_from_slice(full);
-			secp256k1::PublicKey::from_slice(&tagged_full)
+			&tagged_full
 		} else {
-			secp256k1::PublicKey::from_slice(full)
+			full
 		};
-		pubkey.map(|k| Self(k.serialize())).map_err(|_| ())
+		let pubkey = {
+			#[cfg(feature = "std")]
+			{
+				PublicKey::from_slice(&full)
+			}
+			#[cfg(not(feature = "std"))]
+			{
+				VerifyingKey::from_sec1_bytes(&full)
+			}
+		};
```

Comment:
```suggestion
		#[cfg(feature = "std")]
		let pubkey = PublicKey::from_slice(&full);
		#[cfg(not(feature = "std"))]
		let pubkey = VerifyingKey::from_sec1_bytes(&full);
```
This is just personal preference. Feel free to ignore :-D